test_app/tests/lib/redis/test_client.py

```diff
@@ -396,3 +398,89 @@ def test_redis_standalone_removes_cluster_settings():
         rm.assert_called_once
         assert 'host' in rm.call_args.kwargs
         assert 'cluster_error_retry_attempts' not in rm.call_args.kwargs
+
+
+@pytest.mark.parametrize(
+    "the_class",
+    [
+        DABRedis,
+        DABRedisCluster,
+    ],
+)
+def test_redis_clients_have_flushdb_mixin(the_class):
+    assert issubclass(the_class, DABRedisRespectACLFlushMixin)
+
+
+def test_flushdb_warns_about_async(expected_log):
+    args = {'mode': 'standalone'}
+    from ansible_base.lib.redis.client import RedisClientGetter
+
+    with mock.patch('redis.Redis.__init__', return_value=None):
+        client_getter = RedisClientGetter()
+        client = client_getter.get_client('rediss://localhost', **args)
+        client.keys = mock.MagicMock(return_value=[])
+        client.delete = mock.MagicMock(return_value=None)
+        with expected_log('ansible_base.lib.redis.client.logger', 'warning', "DABRedis clients implement an ACL friendly FLUSHDB which can not be async"):
+            client.flushdb(asynchronous=True)
+
+
+class notADABClient(DABRedisRespectACLFlushMixin, Redis):
+    pass
+
+
+def test_flushdb_not_a_dab_client():
+    with mock.patch('redis.Redis.__init__', return_value=None):
+        with pytest.raises(ImproperlyConfigured):
+            client = notADABClient('rediss://localhost')
+            client.flushdb()
+
+
+def test_ensure_flushdb_does_not_die_on_no_permission_exception():
+    args = {'mode': 'standalone'}
+    from redis.exceptions import NoPermissionError
+
+    from ansible_base.lib.redis.client import RedisClientGetter
+
+    with mock.patch('redis.Redis.__init__', return_value=None):
+        client_getter = RedisClientGetter()
+        client = client_getter.get_client('rediss://localhost', **args)
+        client.keys = mock.MagicMock(return_value=[])
+        client.delete = mock.MagicMock(side_effect=NoPermissionError())
+        # This should just not raise if NoPermissionError is raised
+        client.flushdb()
+
+
+def test_redis_cluster_passes_target_nodes():
+    args = {'mode': 'cluster'}
+    from ansible_base.lib.redis.client import RedisClientGetter
+
+    with mock.patch('redis.RedisCluster.__init__', return_value=None):
+        client_getter = RedisClientGetter()
+        client = client_getter.get_client('rediss://localhost', **args)
+        client.keys = mock.MagicMock(return_value=[])
+        client.delete = mock.MagicMock(return_value=None)
+        # This should just not raise if NoPermissionError is raised
+        client.flushdb()
+        client.keys.assert_called_with("*", target_nodes=client.ALL_NODES)
+
+
+def test_flushdb_some_keys_delete_some_do_not(monkeypatch):
```

Comment:
Seems like this test covers `test_ensure_flushdb_does_not_die_on_no_permission_exception` . Maybe delete the other test.

Reply:
Yea, this one is harder because we don't have an actual Redis running with ACLs in the unit tests. The goal was to mock the delete  function so that one key would be deleted and one would not. But in the end, it's just raising the same exception. So I'll just delete this test. It's technically not even really working  because  key_name would be an array of `['key1', 'key2']`. The more I think about it, I don't think this test provides any value to us.